text/0008-weighted-uniform-random-tip-selection/0008-weighted-uniform-random-tip-selection.md

```diff
@@ -0,0 +1,205 @@
+
+# Summary
+
+Weighted Uniform Random Tip Selection on a subset enables a node to perform fast tip-selection to increase transaction throughput.
+The algorithm selects tips which are non-lazy in order to maximize confirmation rate.
+
+# Motivation
+
+Because of the `white-flag` confirmation algorithm, it is no longer necessary to perform complex
+tip-selection which evaluates ledger mutations while walking, therefore, a simpler and better 
+performing algorithm can be used to select tips, which in turn increases overall transaction throughput.
+
+In order to maximize confirmation rate however, the algorithm needs to return tips which are `non-lazy`.
+Non-lazy in this context means, that a tip does not attach to a cone of transactions which is too far
+in the past, as such cone is likely to be already confirmed and therefore does not contribute to the
+rate of newly confirmed transactions when a milestone is issued.
+
+# Detailed design
+
+Definitions:
+* `Direct Approvers` are the set of transactions which directly approve a given transaction.
+* `Approvee` is the directly approved transaction of a given transaction.
+Example: the trunk/branch transactions are the approvees of a given transaction.
+* `Solid` means that the past cone of a given transaction exists in the database.
+* A `tail transaction` is the transaction at index zero of a bundle. Only a tail transaction can be a `tip`.
+* A `valid bundle` is a bundle which is structurally correct and has valid signatures (in case it moves funds).
+* A `tip` is a solid tail transaction of a valid bundle without any approvers. Its past cone contains only valid bundles.
+* A `score` is a scoring  determining the likeliness to select a given `tip`.
+* `Confirmed Root Transactions` defines the set of first seen transactions which are confirmed by a previous milestone 
+when we walk the past cone of a given transaction. The walk stops on confirmed transactions.  
+Yellow = `Confirmed Root Transactions` for PoV transaction, Red = Milestone, Blue = PoV transaction.
+Note that the red marked milestone is also a `confirmed root transaction`.
+![sdf](images/cnf_tx_roots.PNG)
+* `Transaction Snapshot Index (TSI)` defines the index of the milestone which confirmed a given transaction.
+* `Oldest Transaction Root Snapshot Index (OTRSI)` defines the lowest milestone index of a set of
+`Confirmed Root Transactions` of a given transaction.
+* `Youngest Transaction Root Snapshot Index (YTRSI)` defines the highest milestone index of a set of
+`Confirmed Root Transactions` of a given transaction.
+* `Latest Solid Milestone Index (LSMI)` the index of the latest solid milestone.
+* `Below Max Depth (BMD)` defines a threshold value up on which it is decided on whether a transaction is not
+relevant in relation to the recent parts of the Tangle. The current `BMD` for mainnet nodes is 15 milestones, 
+which means that transactions of which their `OTRSI` is more than 15, are "below max depth".

```

Comment:
> of which their `OTRSI` is more than 15, are "below max depth".

Shouldn't it be in relation to LSMI instead of just OTRSI ?

Reply:
Adjusted.